src/middlewared/middlewared/schema.py

```diff
@@ -118,9 +132,9 @@ class Str(EnumMixin, Attribute):
 
     def clean(self, value):
         value = super(Str, self).clean(value)
-        if value is None and not self.required:
-            return self.default
-        if isinstance(value, int):
+        if value is None:
+            return value
+        if isinstance(value, int) and not isinstance(value, bool):
```

Comment:
I don't understand this check (` and not isinstance(value, bool)`) here and further, could you please explain?

Reply:
@themylogin 

    >>> isinstance(False, int)
    True

Reply:
Oh ****